### ISSUE TITLE
using the repeat command (.) after inserting a tab, does nothing

### ISSUE THREAD

**Issue by cloudranger** [VSCodeVim/Vim#2337]
**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):
BUG REPORT

**What happened**:
after inserting a tab, the _._ key cannot be used to repeat the last command (but repeat works for any other character, it seems to be only tab which does not work)

**What did you expect to happen**:
I expected the _._ to insert a tab for me

**How to reproduce it (as minimally and precisely as possible)**:
position to start of a line of text
_i_ _<tab>_ _ESC_ 
position to start of next line
_._ (to repeat last command)
nothing happens.

**Environment**:
- Extension Name: vim
- Extension Version: 0.10.13
- OS Version: Linux x64 4.4.0-38-generic
- VSCode version: 1.19.1

**Comment by parkovski** [comment]
Probably related, number prefixes don't work with tab - `5i<Tab><Esc>` inserts one tab, not five.

**Comment by Chillee** [comment]
So interestingly, it's only if `<tab>` is the only type of action performed. `iabc<tab><esc>.` works as expected.

**Comment by cloudranger** [comment]
Yep, just the tab on its own.  So you can't repeat inserting a tab on its own.  Any other character in there and it works just fine.

**Comment by rnewman** [comment]
For me, `.` is just altogether broken right now. Do anything in insert mode, switch to normal mode with `Esc`, hit `.`, and nothing happens.

**Comment by Chillee** [comment]
@rnewman That's a separate bug, one that I can't replicate at least. Could you open up a new issue, and post any errors you might have in your dev console? (Help > Toggle Developer Tools)

**Comment by jpoon** [comment]
This behaviour seems consistent with vanilla VIM. In VIM, if I go `i<tab><esc>.`, it doesn't re-insert the either on `.`.

However, I can repro the case where `5i<Tab><Esc>` is indeed not supported on this extension.

**Comment by cloudranger** [comment]
Thats interesting.  

I have been using vi and vim for the last 30 years on a variety of unix platforms and it has always worked for me.

I have just tried it now on Ubuntu 16.04 LTS (vim 7.4.1689), Debian 8 (Jessie) (vim 7.4.576) and Mac OS 10.11.6 (vim 7.3) and re-insert of tab works on all of them.   

So what version of vanilla vim are you using and on which platform ?

**Comment by rnewman** [comment]
It's something to do with indent handling. Trailing tab indents are not repeated: it's 

Vim 8.0.1420, Mac, launched with `vim --clean -v`:

* `i<Space><Space><Esc>.` inserts four spaces.
* `i<Tab><Esc>.` does not insert two tabs; only one remains.
* `i<Tab>a<Esc>.` inserts two tabs: `\t\taa`.
* `ia<Tab><Esc>.` inserts `aa\t` (note a single trailing tab — the first one inserted).

**Comment by xconverge** [comment]
It is because we do not handle <tab> the same as a <space>

https://github.com/VSCodeVim/Vim/blob/5fe14b435ba7879d7c2fdd00d69a11a5c0a553e4/src/actions/commands/insert.ts#L218

So we never capture it as a documentChangeEvent, hmm

**Comment by jeremyong** [comment]
Can we just enable that handling? In vim, if you use tab to autocomplete or whatever, the "repeat" command respects the insertion as it should. In fact, overriding this to do anything else breaks the semantics of the dot command altogether (which really should just repeat whatever you just did).

**Comment by jkassis** [comment]
i've noticed that 
```
itypesomestuff<esc>...
```
goes to insert mode, types 'typesomestuff' and repeats as expected

however...
```
itypesomestuff<esc><esc>...
```
does not repeat.

it seems that . repeats the esc and i believe it should not. in neovim, for example, it does not.

**Comment by jkassis** [comment]
ahhh... looks like i remapped escape to also run `:nohl`. maybe this is happening for others? everyone check your mappings?

**Comment by jwlv** [comment]
I am seeing this problem as well.  Really makes it difficult to tab indent a large block of existing code.

**Comment by jeremyong** [comment]
Folks, I recommend moving to the vscode neovim plugin. I keep revisiting this issue every few months and it's not worth it.
